contracts/deploy/015_buyback.js

```diff
@@ -0,0 +1,116 @@
+const {
+  isMainnet,
+  isFork,
+  isRinkeby,
+  isSmokeTest,
+} = require("../test/helpers.js");
+const {
+  log,
+  deployWithConfirmation,
+  withConfirmation,
+  executeProposal,
+  sendProposal,
+} = require("../utils/deploy");
+const { proposeArgs } = require("../utils/governor");
+const { getTxOpts } = require("../utils/tx");
+
+const deployName = "015_buyback_contract";
+
+const runDeployment = async (hre) => {
+  console.log(`Running ${deployName} deployment...`);
+
+  const { governorAddr, deployerAddr } = await hre.getNamedAccounts();
+
+  // Signers
+  const sGovernor = await ethers.provider.getSigner(governorAddr);
+  const sDeployer = await ethers.provider.getSigner(deployerAddr);
+
+  await deployWithConfirmation("Buyback");
+  const cBuyback = await ethers.getContract("Buyback");
+
+  // Initiate transfer of Buyback governance to the governor
+  await withConfirmation(
+    cBuyback
+      .connect(sDeployer)
+      .transferGovernance(governorAddr, await getTxOpts())
+  );
+  log(`Buyback transferGovernance(${governorAddr} called`);
+
+  // Deploy a new VaultCore contract
+  const dVaultCore = await deployWithConfirmation("VaultCore");
+  const cVaultProxy = await ethers.getContract("VaultProxy");
+  const cVaultAdmin = await ethers.getContractAt(
+    "VaultAdmin",
+    cVaultProxy.address
+  );
+
+  // Proposal to:
+  // - Upgrade VaultCore to pick up the Buyback contract integration
+  // - Claim Governance on the Buyback contract
+  // - Set the trustee address to the Buyback contract
+  const propDescription = "Deploy and integrate Buyback contract";
+  const propArgs = await proposeArgs([
+    {
+      contract: cVaultProxy,
+      signature: "upgradeTo(address)",
+      args: [dVaultCore.address],
+    },
+    {
+      contract: cBuyback,
+      signature: "claimGovernance()",
+    },
+    {
+      contract: cVaultAdmin,
+      signature: "setTrusteeAddress(address)",
+      args: [cBuyback.address],
```

Comment:
Should this be done separately, after the governor claimed governance?

Reply:
I don't see a reason to, the whole thing either works or it doesn't, any reason you had in mind?